.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,109 @@
+# This workflow is a mix of:
+# - https://github.com/pola-rs/r-polars/blob/main/.github/workflows/check.yaml
+# - https://github.com/r-lib/actions/blob/v2-branch/examples/pkgdown.yaml
+# - https://squidfunk.github.io/mkdocs-material/publishing-your-site/?h=deploy#material-for-mkdocs
+
+name: mkdocs
+
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
```

Comment:
I think we need to be careful not to push websites by PR.

Reply:
Same as above, we want the CI to trigger on pull requests to see if a specific PR creates an error in the documentation. However, we shouldn't push the changes indeed so we need to add `github.event_name != 'pull_request'` at the end of the CI to avoid committing every time there's a PR